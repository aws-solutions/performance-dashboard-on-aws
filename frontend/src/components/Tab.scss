@forward "../styles/base.scss";

.tab {
  @include u-text("base", "!important");
  background-color: transparent !important;
}
.tab:hover {
  @include u-text("base-dark", "!important");
  background-color: transparent !important;
}
.tab-active {
  @include u-text("base", "!important");
  background-color: transparent !important;
  border-bottom: 3px solid color("base-dark") !important;
}
