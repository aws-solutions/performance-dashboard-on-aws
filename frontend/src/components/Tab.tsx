/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React, { MouseEvent } from "react";
import { useTranslation } from "react-i18next";
import styles from "./Tab.module.scss";

interface Props {
    itemId: string;
    activeTab: string;
    label: string;
    onClick: Function;
    onEnter: Function;
}

function Tab(props: Props) {
    const { t } = useTranslation();
    const onClick = (e: MouseEvent<HTMLElement>) => {
        props.onClick(props.itemId, e.currentTarget);
    };

    const onKeyDown = (e: React.KeyboardEvent<HTMLElement>) => {
        if (e.key === "Enter") {
            props.onEnter(props.itemId, e.currentTarget);
        }
    };

    let className = `${styles.tab} display-inline-block padding-x-2 padding-y-105 text-bold font-sans-md`;

    return (
        <div
            id={`${props.itemId}-tab`}
            aria-controls={`${props.itemId}-panel`}
            aria-selected={props.activeTab === props.itemId}
            role="tab"
            className={className}
            onClick={onClick}
            onKeyDown={onKeyDown}
            tabIndex={props.activeTab === props.itemId ? 0 : -1}
        >
            {props.label}
        </div>
    );
}

export default Tab;
