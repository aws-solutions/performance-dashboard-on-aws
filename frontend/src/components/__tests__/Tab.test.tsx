/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { render, within } from "@testing-library/react";
import Tab from "../Tab";
import { MemoryRouter } from "react-router-dom";

test("renders the Tab component", async () => {
    const wrapper = render(
        <Tab id="tab1" activeTab="tab1" key="Tab 1" label="Tab 1" onClick={() => {}} />,
    );
    expect(wrapper.container).toMatchSnapshot();
});

test("renders the Tab component with default tab 2 selected", async () => {
    const wrapper = render(
        <Tab id="tab1" activeTab="tab2" key="Tab 1" label="Tab 1" onClick={() => {}} />,
    );
    expect(wrapper.container).toMatchSnapshot();
});

test("renders the tab", async () => {
    const { getAllByRole } = render(
        <Tab id="tab1" activeTab="tab2" key="Tab 1" label="Tab 1" onClick={() => {}} />,
        {
            wrapper: MemoryRouter,
        },
    );

    const listItems = getAllByRole("tab");
    expect(listItems).toHaveLength(1);

    listItems.forEach((item, index) => {
        const { getByText } = within(item);
        expect(getByText(`Tab ${index + 1}`)).toBeInTheDocument();
    });
});
