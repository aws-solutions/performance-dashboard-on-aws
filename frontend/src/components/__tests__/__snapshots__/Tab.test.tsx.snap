// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Tab component 1`] = `
<div>
  <div
    aria-controls="undefined-panel"
    aria-selected="false"
    class="tab display-inline-block padding-x-2 padding-y-105 text-bold font-sans-md"
    id="undefined-tab"
    role="tab"
    tabindex="-1"
  >
    Tab 1
  </div>
</div>
`;

exports[`renders the Tab component with default tab 2 selected 1`] = `
<div>
  <div
    aria-controls="undefined-panel"
    aria-selected="false"
    class="tab display-inline-block padding-x-2 padding-y-105 text-bold font-sans-md"
    id="undefined-tab"
    role="tab"
    tabindex="-1"
  >
    Tab 1
  </div>
</div>
`;
